semantic-source/src/Source/Loc.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE DeriveGeneric, DerivingVia, RankNTypes #-}
+module Source.Loc
+( Loc(..)
+, byteRange_
+, Span(Span)
+, Range(Range)
+) where
+
+import Control.DeepSeq (NFData)
+import Data.Hashable (Hashable)
+import Data.Monoid.Generic
+import GHC.Generics (Generic)
+import Prelude hiding (span)
+import Source.Range
+import Source.Span
+
+data Loc = Loc
+  { byteRange :: {-# UNPACK #-} !Range
+  , span      :: {-# UNPACK #-} !Span
```

Comment:
Furthermore, and in keeping with our style guide, I’ve opted to rename its fields to drop the prefix.